modules/consensus/consensusset.go

```diff
@@ -222,6 +228,12 @@ func (cs *ConsensusSet) CurrentBlock() (block types.Block) {
 	return block
 }
 
+// Flush will block until the consensus set has finished all in-progress
+// routines.
+func (cs *ConsensusSet) Flush() error {
+	return cs.tg.Flush()
+}
```

Comment:
It's the only way to get strong reliability on the asynchronous stuff though. Otherwise you make a block and have no idea when it's propagated to the miner, or to the other consensus sets, etc.

I think it's imperfect anyway though.